### PR TITLE
use threading.Lock.acquire's timeout for backpressure rather than sleep

### PR DESCRIPTION
**Pull request by ypsah**
fixes: #649

### CODE REVIEW

File: sh.py [hunk 1]
```diff
@@ -2491,16 +2491,12 @@ def is_alive(self):
         # so essentially what we're doing is, using this lock, checking if
         # we're calling .wait(), and if we are, let .wait() get the exit code
         # and handle the status, otherwise let us do it.
-        acquired = self._wait_lock.acquire(False)
```
Comment (amoffat) [inline sh.py:2494]: There must have been a reason why `blocking=False` and no timeout, but I can't think of it...

**Review by amoffat — APPROVED** [review]

File: sh.py [hunk 2]
```diff
@@ -2491,16 +2491,12 @@ def is_alive(self):
...
+        #
+        # Using a small timeout provides backpressure against code that spams
+        # calls to .is_alive() which may block the main thread from acquiring
+        # the lock otherwise.
+        acquired = self._wait_lock.acquire(timeout=0.1)
```
Comment (JacobHayes) [inline sh.py:2498]: I still had to set this back down to:
```suggestion
        acquired = self._wait_lock.acquire(timeout= 0.00001)
```
as it was before https://github.com/amoffat/sh/commit/e9511de5ca9b1d1bf0b42da51fa40073a73fa4d6.

Given we're using the `acquire` provided timeout, is that fine now?

**Review by JacobHayes — CHANGES_REQUESTED** [review]

**Comment by JacobHayes** [comment]
Ah, a minute too late 😁

**Comment by amoffatgmi** [comment]
There's still time @JacobHayes, it's only merged into the `develop` branch. It won't ship until it lands in the main branch. If you have a proposed change, make it and tag @ypsah

**Pull request merged**